fe/fe-core/src/main/java/org/apache/doris/nereids/processor/post/RecomputeOutputProcessor.java

```diff
@@ -22,13 +22,14 @@
 import org.apache.doris.nereids.trees.plans.physical.AbstractPhysicalPlan;
 
 /**
- * merge consecutive projects
+ * recompute output in logical properties
+ * after join reorder, plan output slot order is changed, recompute them.
  */
-public class RecomputeLogicalPropertiesProcessor extends PlanPostProcessor {
+public class RecomputeOutputProcessor extends PlanPostProcessor {
     @Override
     public Plan visit(Plan plan, CascadesContext ctx) {
         AbstractPhysicalPlan newPlan = (AbstractPhysicalPlan) visitChildren(this, plan, ctx);
-        return ((AbstractPhysicalPlan) newPlan.resetLogicalProperties())
+        return ((AbstractPhysicalPlan) newPlan.reComputeOutput())
```

Comment:
why change this to only recompute output?

Reply:
when we implement this rule, logicalProperty only contains output.
but now we have other information (data trait) which can not be recomputed. So we need to keep datatrait